core/src/test/java/org/wso2/msf4j/PersistentSessionManagerTest.java

```diff
@@ -0,0 +1,96 @@
+import java.util.Map;
+
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertNull;
+
+public class PersistentSessionManagerTest {
+
+    private MicroservicesRunner microservicesRunner;
+    private URI baseURI;
+
+    @BeforeClass
+    public void setup() {
+        baseURI = URI.create(String.format("http://%s:%d", Constants.HOSTNAME, Constants.PORT));
+        microservicesRunner = new MicroservicesRunner(Constants.PORT);
+        microservicesRunner.setSessionManager(new PersistentSessionManager());
+        microservicesRunner.deploy(new SecondService());
+        microservicesRunner.start();
+    }
+
+    @Test
+    public void testSession() throws IOException {
+        HttpURLConnection urlConnection = TestUtil.request(baseURI, "/SecondService/session", "GET");
+        String content = TestUtil.getContent(urlConnection);
+        Map<String, List<String>> headerFields = urlConnection.getHeaderFields();
+        assertEquals("1", content);
+        urlConnection.disconnect();
+
+        urlConnection = TestUtil.request(baseURI, "/SecondService/session", "GET");
+        urlConnection.setRequestProperty("Cookie", headerFields.get("Set-Cookie").get(0));
+        content = TestUtil.getContent(urlConnection);
+        headerFields = urlConnection.getHeaderFields();
+        assertEquals("2", content);
+        urlConnection.disconnect();
+
+        urlConnection = TestUtil.request(baseURI, "/SecondService/session", "GET");
+        //Since the session is invalidated there shouldn't be a cookie
+        assertNull(headerFields.get("Set-Cookie"));
+        content = TestUtil.getContent(urlConnection);
+        assertEquals("1", content);
+        urlConnection.disconnect();
+    }
+
+    @Test
+    public void testRemoveSessionAttribute() throws IOException {
+        HttpURLConnection urlConnection = TestUtil.request(baseURI, "/SecondService/removeSessionAttribute", "GET");
+        String content = TestUtil.getContent(urlConnection);
+        Map<String, List<String>> headerFields = urlConnection.getHeaderFields();
+        assertEquals("1", content);
+        urlConnection.disconnect();
+
+        urlConnection = TestUtil.request(baseURI, "/SecondService/removeSessionAttribute", "GET");
+        urlConnection.setRequestProperty("Cookie", headerFields.get("Set-Cookie").get(0));
+        content = TestUtil.getContent(urlConnection);
+        assertEquals("2", content);
+        urlConnection.disconnect();
+
+        urlConnection = TestUtil.request(baseURI, "/SecondService/removeSessionAttribute", "GET");
+        //Since the session is invalidated there shouldn't be a cookie
+        urlConnection.setRequestProperty("Cookie", headerFields.get("Set-Cookie").get(0));
+        content = TestUtil.getContent(urlConnection);
+        assertEquals("3", content);
+        urlConnection.disconnect();
+    }
+
+    @AfterClass
+    public void clean() {
+        microservicesRunner.stop();
+    }
+}
```

Comment:
We need a line break at the end of the file.

Reply:
Line break is already there at the end of the file